repos/system_upgrade/el7toel8/actors/cloud/grubenvtofile/actor.py

```diff
@@ -0,0 +1,26 @@
+from leapp.actors import Actor
+from leapp.models import HybridImageDetected
+from leapp.tags import FinalizationPhaseTag, IPUWorkflowTag
+from leapp.libraries.actor.grubenvtofile import grubenv_to_file
+
+
+class GrubenvToFile(Actor):
+    """
+    Azure images provided by Red Hat aim for hybrid (BIOS/EFI) functionality,
+    however, currently GRUB is not able to see "grubenv" file if it is a symlink
+    to a different partition (default on EFI with grub2-efi pkg installed) and
+    fails on BIOS systems. This actor converts the symlink to the normal file
+    with the content of grubenv on EFI partition in case the system is using BIOS
+    and running on Azure cloud. This action is reported in the preupgrade phase.
+    """
+
+    name = 'grubenvtofile'
+    consumes = (HybridImageDetected,)
+    produces = ()
+    tags = (FinalizationPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        grubenv_msg = next(self.consume(HybridImageDetected), None)
+
+        if grubenv_msg:
```

Comment:
I guess this is not correct. If you create (produce) the default `HybridImageDetected`, this will be always false. Extend please the model (see suggestion around the mode).

Reply:
@pirat89 could you please explain why it would be false? It is obviously working for me :) Why to complicate it with some field? We either send it or not. The message name is self-explanatory.